### PR TITLE
Increase timeouts and add more flags for tsan builds on RBE

### PR DESCRIPTION
**Pull request by adelez**
CC @nlopezgi

### CODE REVIEW

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ source tools/internal_ci/helper_scripts/prepare_build_linux_rc
 "${KOKORO_GFILE_DIR}/bazel_wrapper.py" \
   --host_jvm_args=-Dbazel.DigestFunction=SHA1 \
   test --jobs="50" \
-  --test_timeout="300,450,1200,3600" \
+  --test_timeout="1500,1500,1500,3600" \
```
Comment (matt-kwong) [inline tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: Just curious: how is the test_timeout argument formatted?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh (hunk 1 above)
Comment (adelez) [reply tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: They correspond to short, moderate, long and external tests.

**Pull request merged**